pkg/io/serialize.go

```diff
@@ -108,6 +112,10 @@ func Decode(reader io.Reader, values ...interface{}) (err error) {
 				return errors.WithMessage(err, "reading length of binary data")
 			}
 
+			// Nothing to be decoded when length is zero.
+			if length == 0 {
```

Comment:
Maybe sometimes we want to unmarshal the empty string?

Reply:
We can still recognize that its an empty data, because we encode/decode the length of data.

Reply:
But, I do not understand your question. What is the case, where we unmarshal the empty string ?

Reply:
I thought. Maybe the serializer sometimes want to write something on the stream to signal that there was an empty string / byte array written. For example, just write uint16(0) as the length without any data following afterwards. Not sure if that is relevant here though.

Reply:
Okay, I get it now. In this particular serializer implementation, it is not relevant, because it does not write anything.

Reply:
@matthiasgeihs

Reply:
OK, I see, and a different serializer implementation could handle it differently.